### PR TITLE
MM-11781: Basic Data Export Command Line.

### PR DESCRIPTION
**Pull request by grundleborg**
#### Summary
Introduces a very basic "bulk export" feature for data that is compatible with the existing "bulk import" feature. The purpose of this PR is to introduce the absolute bare-bones version of bulk export, which we will substantially build upon over the coming releases. It provides only a basic architecture and starting point. The rest will come in future PRs.

This PR includes export for:
* Users
* Teams
* Channels
* Posts in Public or Private channels
* Replies to those posts.

It does not include
* Configuring what is exported - it only exports for the whole server at the moment.
* Many of the non-core properties of the entities it exports.
* Other object types such as DM/GM channels, DM/GM posts, Reactions, Webhook/Bot messages, File Attachements, etc etc etc
* Controlling how "deleted" entities are exported.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-11781

#### Checklist
- [x] Added or updated unit tests (required for all new features)
- [x] Includes text changes and localization file updates

### CODE REVIEW

File: store/sqlstore/channel_store.go [hunk 1]
```diff
@@ -1936,3 +1936,57 @@ func (s SqlChannelStore) ResetLastPostAt() store.StoreChannel {
...
+				Teams.Name as TeamName,
+				Schemes.Name as SchemeName
+			FROM Channels
+			INNER JOIN
+				Teams ON Channels.TeamId = Teams.Id
+			LEFT JOIN
+				Schemes ON Channels.SchemeId = Schemes.Id
+			WHERE
+				Channels.Id > :AfterId
+				AND Channels.Type IN ('O', 'P')
```
Comment (jespino) [inline store/sqlstore/channel_store.go]: I miss here the export of direct channels and group channels, I don't know if is on purpose.

Comment (grundleborg) [reply store/sqlstore/channel_store.go]: Yeah, it's intentional. The data structure for them is different, so they need to come out in a different query. I haven't implemented that in this PR as the idea was to be as minimal as possible - we can add them later.

File: store/sqlstore/post_store.go [hunk 2]
```diff
@@ -1308,3 +1308,68 @@ func (s *SqlPostStore) GetMaxPostSize() store.StoreChannel {
 		result.Data = s.maxPostSizeCached
 	})
 }
+
+func (s *SqlPostStore) GetParentsAfterForExport(limit int, afterId string) store.StoreChannel {
```
Comment (mkraft) [inline store/sqlstore/post_store.go]: Names is inconsistent with `GetAllChannelsForExportAfter`.

**Comment by grundleborg** [comment]
@mkraft made the suggested changes, thanks.

**Pull request merged**